### PR TITLE
Update kafka.go to fix typo and update config for external Kafka Connection in SASL mode

### PR DESCRIPTION
**Pull request by SahilDevgon**

### CODE REVIEW

File: pkg/external/kafka.go [hunk 1]
```diff
@@ -108,7 +108,8 @@ func CheckKafka(conf CheckKafkaConfig) error {
 		ctx, cancel := context.WithTimeout(context.Background(), DependencyCheckTimeout)
 		defer cancel()
 		err := r.SetOffsetAt(ctx, time.Now())
-		return errors.Wrap(err, "check consume offset from borker failed")
+		# Fixing typo in err meesage
```
Comment (haorenfsa) [inline pkg/external/kafka.go:111]: this comment is not necessary

**Review by haorenfsa — CHANGES_REQUESTED** [review]

**Comment by haorenfsa** [comment]
Hi @SahilDevgon, before we can merge your patch, please sign off your commits. check this doc for detail: https://github.com/apps/dco

File: pkg/external/kafka.go (hunk 1 above)
Comment (SahilDevgon) [reply pkg/external/kafka.go]: comment is removed

**Review by SahilDevgon — COMMENTED** [review]
Additional comment is removed as requested

**Comment by haorenfsa** [comment]
about CI failure: the v2.3.6 milvus image has critical bugs, so they deleted this tag to avoid being used.

File: docs/administration/manage-dependencies/message-storage.md [hunk 2]
```diff
@@ -238,8 +238,8 @@ metadata:
 spec:
   config:
     kafka:
-      # securityPolicy supports: PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL 
-      securityPolicy: PLAINTEXT
+      # SecurityProtocol supports: PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL 
+      SecurityProtocol: PLAINTEXT
```
Comment (haorenfsa) [inline docs/administration/manage-dependencies/message-storage.md:242]: Another catch XD, but the fields in `spec.config` are capital sensitive, see:
https://github.com/milvus-io/milvus/blob/v2.3.7/pkg/util/paramtable/service_param.go#L688

**Review by haorenfsa — CHANGES_REQUESTED** [review]

**Review by SahilDevgon — COMMENTED** [review]
Fixed the case-sensitivity issue

**Review by haorenfsa — APPROVED** [review]

**Pull request merged**